pip/assemble.py

```diff
@@ -58,20 +58,21 @@ def create_init_files(directory):
 pkg_dir = tempfile.mkdtemp()
 
 
-for f in args.files:
-    fn = f
-    for _imp in args.imports:
-        match = _imp.match(fn)
-        if match:
-            fn = match.group('fn')
-            break
-    try:
-        e = os.path.join(pkg_dir, os.path.dirname(fn))
-        os.makedirs(e)
-    except OSError:
-        # directory already exists
-        pass
-    shutil.copy(f, os.path.join(pkg_dir, fn))
+if args.files:
```

Comment:
Is there ever a scenario in which `args.files` would be legitimately empty? Does it make sense to assemble a Pip package with no Python sources?

If so, we should keep this change; if not, it would be better to raise an error when `args.files` is None or empty.

Reply:
I think it's technically allowed in the wheel spec, it's likely undesirable from a user's perspective. The error that was thrown didn't make it clear the cause was missing all input files. That might be the better route to take here actually. Something like
```py
if not args.files:
    raise Exception("Cannot create an archive without any files")
```

Reply:
Agreed - let's add that.